packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -215,30 +215,40 @@ export function useInteractionModality(): Modality {
   return modality;
 }
 
+/**
+ * If this is attached to text input component, return if the event is a focus event (Tab/Escape keys pressed) so that
+ * focus visible style can be properly set.
+ */
+export function isKeyboardFocusEvent(isTextInput: boolean, modality: Modality, e: HandlerEvent) {
```

Comment:
Can we keep this function internal? Could `useFocusVisibleEmitter` should do the filtering internally rather than within `useFocusRing`?